presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -65,17 +82,34 @@ public LongDictionaryProvider(InputStreamSources dictionaryStreamSources)
      * @param items number of items expected in the dictionary.
      * @return The DictionaryResult contains two parts:
      * 1) the final dictionary buffer object. Different from the input dictionary buffer if the input
-     *    dictionary buffer is expanded or that the method returns a shared dictionary.
+     * dictionary buffer is expanded or that the method returns a shared dictionary.
      * 2) whether the caller will be the owner of the dictionary, for the purpose of memory accounting.
-     *    Callers own all non-shared dictionaries, and only the first caller of the shared dictionary
-     *    is the owner.
+     * Callers own all non-shared dictionaries, and only the first caller of the shared dictionary
+     * is the owner.
      * @throws IOException
      */
     public DictionaryResult getDictionary(StreamDescriptor streamDescriptor, long[] dictionary, int items)
             throws IOException
     {
-        InputStreamSource<LongInputStream> dictionaryDataStream = dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class);
-        return loadDictionary(streamDescriptor, requireNonNull(dictionaryDataStream), dictionary, items);
+        InputStreamSource<LongInputStream> dictionaryDataStream = requireNonNull(dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class));
+
+        if (streamDescriptor.getSequence() == 0 || dictionaryDataStream.openStream() != null) {
+            return requireNonNull(loadDictionary(streamDescriptor, dictionaryDataStream, dictionary, items));
```

Comment:
I see 2 logic changes in the new code.

1. Addition of the condiition dictionaryDataStream.openStream() != null.
2. requireNonNull on the return value.

If the logic change has to be minimized. 

I would propose the following code to minimize the combinations. This ensures that when the stream is directly found, it uses it. 

if (dictionaryDataStream.openStream() != null) {
 return loadDictionary.
}

checkArgument( sequence != DEFAULT_SEQUENCE_ID);

Reply:
That's fair enough. I was keeping this branching logic as similar to bbio as possible, though there's no production code that relies on bypassing shared dictionary lookup through sequence 0.

Reply:
Can we change the code to the following logic.

if (dictionaryDataStream.openStream() != null) {
return loadDictionary.
}

checkArgument( sequence != DEFAULT_SEQUENCE_ID);